nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/vls.dm

```diff
@@ -313,12 +318,26 @@
 /obj/structure/overmap/proc/handle_autonomous_targeting()
 	if(flak_battery_amount >= 1)
 		handle_flak()
-	if(!weapon_types[FIRE_MODE_AMS])
-		return FALSE
-	var/datum/ship_weapon/AMS = weapon_types[FIRE_MODE_AMS]
+
+	// Get all weapons designated as autonomous and prepare to fire 
+	var/list/automated_weapons = list()
+	if ( weapon_types )
+		for( var/item in weapon_types )
+			if ( item )
+				var/datum/ship_weapon/W = item
+				if ( W.autonomous )
+					if ( W.can_fire() ) // Is it loaded? Is it enabled? Is it not broken? 
+						automated_weapons += W // Weapons are prioritized by the order they are defined in overmap.dm. Only the first priority weapon will be fired for each mode 
+	
 	for(var/datum/ams_mode/AMM in ams_modes)
 		if(AMM.enabled)
-			AMM.handle_autonomy(src, AMS)
+			for ( var/datum/ship_weapon/W in automated_weapons ) 
```

Comment:
Interesting approach, maybe a check for it not having any permitted AMS modes to save a bit of CPU?